lib/apiservers/portlayer/restapi/handlers/interaction_handlers.go

```diff
@@ -339,17 +340,70 @@ type FlushingReader struct {
 	io.Reader
 	io.WriterTo
 
-	flusher GenericFlusher
+	flusher   GenericFlusher
+	initBytes []byte
 }
 
 func NewFlushingReader(rdr io.Reader) *FlushingReader {
-	return &FlushingReader{Reader: rdr, flusher: nil}
+	return &FlushingReader{Reader: rdr, flusher: nil, initBytes: nil}
+}
+
+func NewFlushingReaderWithInitBytes(rdr io.Reader, initBytes []byte) *FlushingReader {
+	return &FlushingReader{Reader: rdr, flusher: nil, initBytes: initBytes}
 }
 
 func (d *FlushingReader) AddFlusher(flusher GenericFlusher) {
 	d.flusher = flusher
 }
 
+// readDetectInit() is used by WriteTo() which is used by io.Copy.  It attempts
+// to detect a init byte buffer.  If it finds that init byte sequence, it is
+// ignored.  This reader does not care about the init sequeunce.  The init sequence
+// maybe used by the higher level interaction, which in this case is the Swagger
+// establishing initial connection for stdin.
+func (d *FlushingReader) readDetectInit(buf []byte) (int, error) {
+	nr, er := d.Read(buf)
```

Comment:
Sorry, it bugs me when I'm sure there's a clear and more concise way of writing something, so:
```

// fast path - len(nil) return 0
if len(d.initBytes) == 0 {
    return d.Read(buf)
}

initLen := len(d.initBytes)
total := 0
for total < initLen {
    nr, er := d.Read(buf) 
    total += nr

    if bytes.Compare(d.initBytes[0:total], buf[0:total]) != 0 {
        // First bytes aren't part of init bytes so client must not be
        // the docker personality so break and ignore looking for the
        // init bytes.
        log.Debugf("Did not find primer bytes, stopping watch")
        return total, er
    }

    if er != nil && total < initLen{
        // you're right - we've got no idea if these were init bytes or
        // not at this point so don't return data
        return 0, err
    }    
}

// would have returned in the compare clause if not matching init bytes
copy(buf[0:], buf[initLen:])
log.Debugf("Found primer bytes, port layer client might be personality server")

// no risk of returning <0
return total - initLen, er
```